server/v2/commands.go

```diff
@@ -52,7 +52,17 @@ func Commands[AppT AppI[T], T transaction.Tx](
 		Use:   "start",
 		Short: "Run the application",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			v := GetViperFromCmd(cmd)
+			home, err := cmd.Flags().GetString(FlagHome)
+			if err != nil {
+				return err
+			}
+			// Read server config from home flag
+			// then bind cmd flags later
+			v, err := ReadConfig(filepath.Join(home, "config"))
```

Comment:
We've read the config earlier, here: https://github.com/cosmos/cosmos-sdk/blob/9df6019de6ee7999fe9864bac836deb2f36dd44a/server/v2/commands.go#L149
We set it in the command context and set the command context to the root command.
This should not be needed, and we should be able to get it directly from the command (as it was)